src/interp.rs

```diff
@@ -110,21 +109,58 @@ impl<'tcx> TyVal<'tcx> {
             TypeVariants::TyInt(int_ty) => match int_ty {
                 IntTy::I32 => {
                     let v = LittleEndian::read_i32(bytes.as_slice());
-                    let val = Value::Int(v as u128);
-                    return TyVal::new(ty, val);
+                    return Value::Int(v as u128);
                 },
                 _ => unimplemented!()
             },
             TypeVariants::TyBool => {
                 assert_eq!(bytes.len(), 1);
                 let val = bytes[0] == 1;
-                TyVal::new(ty, Value::Bool(val))
+                return Value::Bool(val)
             },
+            TypeVariants::TyTuple(ref elem_types, ..) => {
+                let mut start = 0;
+                let mut vals: Vec<Value> = Vec::new();
+                for ty in elem_types.iter() {
+                    let end = start + size_of(ty);
+                    let sub = bytes[start..end].to_vec();
+                    vals.push(Value::from_bytes(sub, ty));
+                    start = end;
+                }
+                return Value::Aggregate(vals);
+            }
             _ => unimplemented!()
         }
     }
 }
 
+/// A Rust MIR value and its type. Used to help de/serialize values to bytes for
```

Comment:
This stuff just moved, correct?

Reply:
Yes, I moved the `to_bytes` and `from_bytes` methods to be implemented on the `Value` enum instead of the `TyVal` struct.